src/state.rs

```diff
@@ -44,24 +44,29 @@ enum DkgCurrentState {
     WaitingForMoreParts,
 }
 
-impl<R: bls::rand::RngCore + Clone> DkgState<R> {
+impl<R: bls::rand::RngCore> DkgState<R> {
     pub fn new(
         our_id: NodeId,
         secret_key: SecretKey,
         pub_keys: BTreeMap<NodeId, PublicKey>,
         threshold: usize,
-        rng: &mut R,
+        mut rng: R,
```

Comment:
If you had a deterministic Rng, you'd still run into issue with this API, this pushes the clone up to the caller.

Do we need to hold on to the RNG in our struct? If we took the RNG as an argument to each method that needs access to an RNG, we wouldn't have this problem

Reply:
We don't *have* to, but the alternative is to pass an `rng` every time we `handle_signed_vote` and it will be unused most of the time.

Reply:
Yeah I'd say so, this way we can avoid this cloning issue both in the sn_sdkg code base and any user of this library.